api/v1beta3/ratelimitpolicy_types.go

```diff
@@ -141,7 +141,7 @@ type RateLimitPolicySpec struct {
 	// TargetRef identifies an API object to apply policy to.
 	// +kubebuilder:validation:XValidation:rule="self.group == 'gateway.networking.k8s.io'",message="Invalid targetRef.group. The only supported value is 'gateway.networking.k8s.io'"
 	// +kubebuilder:validation:XValidation:rule="self.kind == 'HTTPRoute' || self.kind == 'Gateway'",message="Invalid targetRef.kind. The only supported values are 'HTTPRoute' and 'Gateway'"
-	TargetRef gatewayapiv1alpha2.LocalPolicyTargetReference `json:"targetRef"`
+	TargetRef gatewayapiv1alpha2.LocalPolicyTargetReferenceWithSectionName `json:"targetRef"`
```

Comment:
Do you mind keeping this one commit out if this PR please? It's done in https://github.com/Kuadrant/kuadrant-operator/pull/893 where it makes more sense IMO, because the functionality for it is also implemented there.

Reply:
I need that to implement the removal of routeSelectors. It's part of my assignment on https://github.com/Kuadrant/kuadrant-operator/issues/810

Reply:
Please. Don't do that either. It's also removed in #893, along with the meaning of that.

Reply:
I am confused. I thought we agreed on doing the version bumping first together with the route selector removal. That's a change big enough to deserve it's own PR. Then in another PR, the refactor on the state of the world controller. 

I think it is reasonable and we should stick to the plan.

Reply:
> adding the sectionName has no meaning without the task the computes the effect RLPs.

Completely disagree. Adding the sectionName and removing the routeSelectors is itself the point of this PR. And it can be done without computing the effective policies and without refactor to to the state of the world controller

Reply:
@eguzki, I see your point, but we are doing the same work twice. The work in #893 is done already, and that branch targets this one in case you want to merge it in one go.

Redoing the same thing in this PR is a waste of your time IMO, first because it's done in there and it will all be part of the same change anyway, but more importantly to me because here it would be a change to the API without the counterpart function that uses this field.

That said, if you really want to work on some part of this particular change still in this PR, the only thing I ask is please do not add `sectionName` just yet. You can remove `routeSelectors` and all the logic related to it. I'm fine with rebasing mine. Np.